pages/installation.md

```diff
@@ -8,25 +8,302 @@ permalink: /installation/
+
+### RabbitMQ
+
+Assuming that you have RabbitMQ install correctly (and added to PATH), you need add yourself a user to rabbitmq.
+
+##### Create a user
+
+```
+sudo rabbitmqctl add_user oop somepassword
+sudo rabbitmqctl list_permissions --vhost /
+sudo rabbitmqctl set_permissions -p / oop ".*" ".*" ".*"
+```
+
+Ensure this is reflected in `OOP_AMQP_ADDRESS` connection string environment variable.
+
+
+### .global-env
+
+A file including all environment variables will need to be created outside of any of the microservice directories. A symlink will need to be created in each directory to mirror .global-env, the command for this will be included in the instructions for each microservice individually.
+
+##### `.global-env`
+```
+NODE_ENV=production
+RAILS_ENV=production
+
+# All repos
+OOP_AMQP_ADDRESS=amqp://oop:somepasssword@localhost
+RABBITMQ_URL=amqp://oop:somepassword@localhost
+OOP_ERROR_EXCHANGE_NAME=oop.errors
+OOP_JSON_ERROR_Q=oop.errors.json
+OOP_EXCHANGE_NAME=oop
+OOP_QUEUE_PREFETCH_LIMIT=5
+
+# Gateway
+OOP_GATEWAY_OUTPUT_Q=oop.noauth.raw_messages
+OOP_LISTEN_PORT=3000
+OOP_DB_ADDRESS=http://admin:admin@localhost:5984
+
+# Authenticator
+OOP_AUTHENTICATOR_INPUT_Q=oop.noauth.raw_messages
+OOP_AUTHENTICATOR_OUTPUT_Q=oop.hasauth.messages
+OOP_CORE_TOKEN=foobar
+OOP_CORE_DEVICE_UPDATE_EXCHANGE=oop.core.devices
+
+# Tempr
+OOP_TEMPR_INPUT_Q=oop.hasauth.messages
+OOP_TEMPR_OUTPUT_Q=oop.hasauth.temprs
+OOP_CORE_API_URL=http://{CORE_IP_ADDRESS}:9001/services/v1
+
+# Scheduler
+OOP_SCHEDULER_OUTPUT_Q=oop.hasauth.messages
+
+# Renderer
+OOP_RENDERER_INPUT_Q=oop.hasauth.temprs
+OOP_ENDPOINTS_EXCHANGE_NAME=oop.endpoints
+OOP_ENDPOINT_Q=oop.endpoints
+
+# Endpoints HTTP
+OOP_ENDPOINTS_HTTP_OUTPUT_Q=oop.hasauth.relay
+OOP_REQUEST_TIMEOUT=1000
+OOP_ENDPOINTS_HTTP_MAX_RETRIES=3
+
+# Relay
+OOP_RELAY_INPUT_Q=oop.hasauth.relay
+OOP_RECURSIVE_TEMPR_Q=oop.hasauth.temprs
+OOP_CORE_RESPONSE_Q=oop.core.transmissions
+
+# Core
+OOP_AMQP_ADDRESS=amqp://localhost
+OOP_RENDERER_PATH=/path/to/renderer/
+
+OOP_CORE_SCHEME=http://
+OOP_CORE_PORT=9001
+OOP_CORE_PATH=/
+
+PORT=9001
+
+OOP_CORE_INTERFACE_SCHEME=http://
+OOP_CORE_INTERFACE_PORT=80
+OOP_CORE_INTERFACE_PATH=/
+OOP_CORE_FROM_ADDRESS=example@mail.co.uk
+
+# Mail setup
+OOP_CORE_FROM_ADDRESS=
+OOP_CORE_SMTP_ADDRESS=
+OOP_CORE_SMTP_PORT=25
+OOP_CORE_SMTP_DOMAIN=
+OOP_CORE_SMTP_USER_NAME=
+OOP_CORE_SMTP_PASSWORD=
+OOP_CORE_SMTP_AUTHENTICATION=
+OOP_CORE_SMTP_ENABLE_STARTTLS_AUTO=
+
+DATABASE_HOST=localhost
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=password
+POSTGRES_DB=oop_core_development
+
+SECRET_KEY_BASE={SECRETKEY}
+```
+
+### oop-core
+
+Clone the application `git clone http://github.com/open-interop/oop-core.git`
+In the app directory install all dependencies using  `bundle install`
+Create a symlink to the .env file created earlier - `ln -nfs /path/to/.global-env /path/to/oop-core/.env`
+Ensure all variables are set up correctly in the config directory:
+
+###### `database.yml`
```

Comment:
should I add something about the DATABASE_URL env variable that can be used to replace this? or is that not a feature we want people to use?

Reply:
You can do. But, let's not block this release.